### PR TITLE
Change date format to be globally accessible

### LINKED ISSUE #2820: Change Date Formatting to be Globally Accessible
In both the app and the marketing site, we are formatting dates as "mm/dd/yyyy". This is not globally accessible, as in come countries this date would be read as "dd/mm/yyyy". 

We should change all instances where we show dates to something that is more globally understood. The easiest way to do this is to write out the month. I suggest that we can abbreviate the months as follows:

Jan, Feb, Mar, Apr, May, June, July, Aug, Sep, Oct, Nov, Dec

And we will format the date as follows:

Sep 10, 2017  --- with no decimal at the end of the abbreviated month.

### PR DESCRIPTION
**Pull request by singingwolfboy**
#### What are the relevant tickets?
Fixes #2820

#### What's this PR do?
Changes the dashboard format string for formatting dates. Does _not_ change how months are formatted, e.g. for education and employment.

#### How should this be manually tested?
Verify that dates are formatted in the format described in #2820.

date format

### CODE REVIEW

**Comment by roberthouse54** [comment]
Can we make the same change on the Marketing page for Programs. Also the home page (if any). I know there are dates on the Program page.

**Comment by roberthouse54** [comment]
There are dates in the Courses box on the Program page: (see below)
dates

**Comment by pdpinch** [comment]
@roberthouse54 I think we should address the program page in a separate PR, since it's completely different code.

File: static/js/components/dashboard/FinancialAidCard_test.js [hunk 1]
```diff
@@ -122,7 +122,7 @@ describe("FinancialAidCard", () => {
       program.financial_aid_user_info.has_user_applied = false;
       let wrapper = renderCard({ program });
       wrapper.find('.full-price').simulate('click');
-      assert.ok(setSkipDialogStub.calledWith(true), 'Dialog should get opened');
+      sinon.assert.calledWith(setSkipDialogStub, true);
```
Comment (aliceriot) [inline static/js/components/dashboard/FinancialAidCard_test.js]: I think it's nice to have the `assert.ok` message - if you break a test it gives you quick feedback about what didn't happen.

Comment (aliceriot) [inline static/js/components/dashboard/FinancialAidCard_test.js]: could we leave the `assert.ok` calls for stubs as-is?

Comment (singingwolfboy) [inline static/js/components/dashboard/FinancialAidCard_test.js]: Are you talking about the customizable message? [Sinon's assertions](http://sinonjs.org/releases/v1.17.7/assertions/) provide error messages that are also very helpful, although perhaps a customized error message should be _more_ helpful. I'd be happy to revert the changes where I removed a customized error message, but Sinon's default error messages are going to be more helpful and informative than the default `assert.ok` error message.

Is that OK? Revert the changes where I'm removing a customized error message, but keep the changes where there was no customized error message to begin with?

Comment (aliceriot) [inline static/js/components/dashboard/FinancialAidCard_test.js]: Yes, that's what I was suggesting.

Comment (singingwolfboy) [inline static/js/components/dashboard/FinancialAidCard_test.js]: Done!

**Comment by singingwolfboy** [comment]
@pdpinch @roberthouse54: yes, I already made that change in this pull request. 😸 
changed date formatting on program page

**Comment by roberthouse54** [comment]
Cool. Thanks.

**Pull request merged**